src/main/java/qna/service/QnaService.java

```diff
@@ -34,25 +30,10 @@ public Question findQuestionById(Long id) {
 
     @Transactional
     public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteException {
-        Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = answerRepository.findByQuestionIdAndDeletedFalse(questionId);
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, LocalDateTime.now()).deleteBy(question.getWriter()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), LocalDateTime.now()).deleteBy(answer.getWriter()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+        final Question question = findQuestionById(questionId);
+        final Answers answers = new Answers(answerRepository.findByQuestionIdAndDeletedFalse(questionId));
```

Comment:
테스트 코드 추가 반영 👍 
Mock 테스트뿐만 아니라 JPA 관점에서 추가 검증을 해주셨기에
Answer는 Question을 통해 조회를 할 수 있기에 Service는 Answers를 몰라도 될 것 같아요 😄 
```suggestion
        final Question question = findQuestionById(questionId);
```

Reply:
추가적인 리뷰 감사드립니다 👍 
JPA 과제 관련 리뷰 감사합니다 💐